File: Ruben - Modulo 2/prJarras/src/Jarras/Jarra.java
package Jarras;

public class Jarra {
    int contenido;
    final int capacidad;

    public Jarra(int capacidad){
        this.capacidad = capacidad;
        contenido = 0;
    }

    public int capacidad() { return capacidad; }
    public int contenido() { return contenido; }
    public void llena() { contenido = capacidad; }
    public void vacia() { contenido = 0; }
    public void llenaDesde (Jarra j){
        while (this.contenido < this.capacidad && j.contenido > 0){
            j.contenido--;
            this.contenido++;
        }
    }
    @Override
    public String toString() {
        return "(" + contenido + " / " + capacidad + ")";
    }
}

File: Ruben - Modulo 2/prJarras/src/JarrasKT/Jarra.kt
package JarrasKT

class Jarra(private val capacidad: Int) {
    private var contenido: Int

    init {
        contenido = 0
    }

    fun capacidad() = capacidad

    fun contenido() = contenido

    fun llena() {
        contenido = capacidad
    }

    fun vacia() {
        contenido = 0
    }

    fun llenaDesde(j: Jarra) {
        while (this.contenido < this.capacidad && j.contenido > 0) {
            j.contenido--
            this.contenido++
        }
    }

    override fun toString() = "($contenido / $capacidad)"
}